mono_repo/lib/src/commands/github/github_yaml.dart

```diff
@@ -9,27 +9,80 @@ import 'package:pub_semver/pub_semver.dart';
 import '../../ci_shared.dart';
 import '../../package_config.dart';
 import '../../root_config.dart';
+import '../../user_exception.dart';
 import '../../yaml.dart';
 import '../ci_script/generate.dart';
-import '../shared.dart';
+import 'self_validate_job.dart';
 
-String generateGitHubYml(
+Map<String, String> generateGitHubYml(
   RootConfig rootConfig,
   Map<String, String> commandsToKeys,
 ) {
-  final jobs = rootConfig.expand((config) => config.jobs);
+  final jobs = rootConfig.expand((config) => config.jobs).toList();
 
-  final jobList = Map.fromEntries([
-    if (rootConfig.monoConfig.selfValidateStage != null)
-      _selfValidateTaskConfig(),
-    ..._listJobs(jobs, commandsToKeys, rootConfig.monoConfig.mergeStages),
-  ]);
+  final selfValidateStage = rootConfig.monoConfig.selfValidateStage;
+  if (selfValidateStage != null) {
+    jobs.add(SelfValidateJob(selfValidateStage));
+  }
+
+  final allJobStages = jobs.map((e) => e.stageName).toSet();
+
+  final output = <String, String>{};
+
+  void doTheWork(
```

Comment:
can we get a better method name here? what work?

Reply:
yup